src/d3d9/d3d9_mem.cpp

```diff
@@ -0,0 +1,350 @@
+#include "d3d9_mem.h"
+#include "../util/util_string.h"
+#include "../util/util_math.h"
+#include "../util/log/log.h"
+#include "../util/util_likely.h"
+#include <utility>
+
+#ifdef D3D9_ALLOW_UNMAPPING
+#include <sysinfoapi.h>
+#endif
+
+namespace dxvk {
+
+#ifdef D3D9_ALLOW_UNMAPPING
+  D3D9MemoryAllocator::D3D9MemoryAllocator() {
+    SYSTEM_INFO sysInfo;
+    GetSystemInfo(&sysInfo);
+    m_allocationGranularity = sysInfo.dwAllocationGranularity;
+  }
+
+  D3D9Memory D3D9MemoryAllocator::Alloc(uint32_t Size) {
+    std::lock_guard<dxvk::mutex> lock(m_mutex);
+
+    uint32_t alignedSize = align(Size, 64);
```

Comment:
64 == `CACHE_LINE_SIZE` or is there any other reason for this?

Reply:
Yes, basically just avoid cursed alignments.

Reply:
Please use the constant in that case so we know what this is for.

Reply:
Already done.